cypress/e2e/flow-builder/undo-redo/actions-on-block.cy.js

```diff
@@ -1,64 +1,88 @@
+      })
+    })
+  })
+
+  it('can undo/redo duplicating a block', () => {
+    cy.addBlock(['Content', 'Message']).then((blockUuid) => {
+      cy.get(`[data-cy="block--${blockUuid}"]`).should('exist')
+
+      cy.duplicateBlock(blockUuid).then((duplicatedBlockUuid) => {
+        cy.get(`[data-cy="block--${duplicatedBlockUuid}"]`).should('exist')
 
-      // TODO in CORE-664: enable the test after fixing the bug
-      // Block deletion, scenario1: Delete the block right away, with just one block in the snapshot
-      // cy.deleteBlock(messageBlockUuid).then(() => {
-      //   cy.get(`[data-cy="block--${messageBlockUuid}"]`).should('not.exist')
-      //   cy.undo()
-      //   cy.get(`[data-cy="block--${messageBlockUuid}"]`).should('exist')
-      //   cy.redo()
-      //   cy.get(`[data-cy="block--${messageBlockUuid}"]`).should('not.exist')
-      //
-      //   // TODO in CORE-664: bring next tests in this line once the bug is fixed
-      // })
-
-      // #### Duplicate that block
-      let duplicatedMessageBlockUuid = null
-      cy.duplicateBlock(messageBlockUuid).then((blockUuid) => {
-        duplicatedMessageBlockUuid = blockUuid
-        cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('exist')
         cy.undo()
-        cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('not.exist')
+        cy.get(`[data-cy="block--${duplicatedBlockUuid}"]`).should('not.exist')
+
         cy.redo()
-        cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('exist')
-
-        // #### Duplicate multiple blocks
-        cy.addBlock(['Content', 'Numeric Response']).then((blockUuid2) => {
-          cy.duplicateMultipleBlocks([duplicatedMessageBlockUuid, blockUuid2]).then((duplicatedBlocks) => {
-            cy.get(`[data-cy="block--${duplicatedBlocks[0]}"]`).should('exist')
-            cy.get(`[data-cy="block--${duplicatedBlocks[1]}"]`).should('exist')
-            cy.undo()
-            cy.get(`[data-cy="block--${duplicatedBlocks[0]}"]`).should('not.exist')
-            cy.get(`[data-cy="block--${duplicatedBlocks[1]}"]`).should('not.exist')
-            cy.redo()
-            cy.get(`[data-cy="block--${duplicatedBlocks[0]}"]`).should('exist')
-            cy.get(`[data-cy="block--${duplicatedBlocks[1]}"]`).should('exist')
-
-            // #### Block deletion, scenario2: Delete a block, after we have few snapshots
-            cy.deleteBlock(duplicatedMessageBlockUuid).then(() => {
-              cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('not.exist')
-              cy.undo()
-              cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('exist')
-              cy.redo()
-              cy.get(`[data-cy="block--${duplicatedMessageBlockUuid}"]`).should('not.exist')
-            })
-          })
+        cy.get(`[data-cy="block--${duplicatedBlockUuid}"]`).should('exist')
+      })
+    })
+  })
+
+  it('can undo/redo duplicating multiple blocks', () => {
```

Comment:
@object-required , thanks for taking time to refactor the spec to make it more easier to read 🙏

Reply:
+1